### ISSUE TITLE
Better prompt messages for YubiKey 5C

### ISSUE THREAD

**Issue by karamellpelle** [latchset/pkcs11-provider#410]
Thanks for creating this necessary project! I have a YubiKey 5C, and I am able to sign using the YubiKey. These are my settings in _/etc/ssl/openssl.cnf_:

~~~cnf
[provider_sect]
default = default_sect
pkcs11  = pkcs11_sect

[default_sect]
# important to activate default provider if we add additional providers! 
activate = 1

[pkcs11_sect]
activate           = 1
module             = /usr/lib64/ossl-modules/pkcs11.so
# Yubico's libykcs11.so makes it possible to get access to Retired Key data objects:
pkcs11-module-path = /usr/lib/libykcs11.so
pkcs11-module-cache-pins = cache
~~~

To sign a CSR _request.csr_ and create _certificate.crt_ I use for example: 
~~~bash
openssl x509 -req -in request.csr -extfile config/piv.cnf -extensions sign -CA 'pkcs11:id=%05' -CAkey 'pkcs11:id=%05' -sha512 -days 800 -out certificate.crt` 
~~~
where _id=%05_ was found using `pkcs11-tool --module /usr/lib64/libykcs11.so -O` (which is my Root CA added  into Slot 82 (Retired Key 1) using the _ykman_ command) (_piv.cnf_ is a file with my sign settings like [x509v3 keys](https://www.openssl.org/docs/manmaster/man5/x509v3_config.html)). Then I will be given the following prompt:

~~~
Enter pass phrase for PKCS#11 Token (Slot 0 - Yubico YubiKey OTP+FIDO+CCID 00 00):
~~~

If I remove the line `pkcs11-module-cache-pins = cache` in _openssl.cnf_, I get two prompts:

~~~
Enter pass phrase for PKCS#11 Token (Slot 0 - Yubico YubiKey OTP+FIDO+CCID 00 00):
Enter PIN for PKCS#11 Token (Slot 0 - Yubico YubiKey OTP+FIDO+CCID 00 00):
~~~

For each prompt, I need to type my PIN. Then the command stops, but the YubiKey starts blinking, and I have to touch the YubiKey. After that, the CSR gets signed (which is very nice!). But in the beginning I didn't know what "pass phrase" meant  (but tried my PIN - correct), and found out after a while that the command stopped because it was waiting for a touch on the YubiKey.

## Expected behavior
It would be an improvement if you can give better prompt messages:
  * Use the word "PIN" instead of "pass phrase"
  * Print "Touch the YubiKey" (or something similar)

Not sure if these prompt messages are given by OpenSSL or this _pkcs11-provider_ project.

## Additional context
FYI, here is my _piv.cnf_:
~~~cnf
[ req ]
utf8                   = yes
distinguished_name     = req_dn
prompt                 = no

[ req_dn ]
commonName             = James Bond

# PIV.9a
[ auth ]
basicConstraints       = CA:FALSE
subjectKeyIdentifier   = hash
authorityKeyIdentifier = keyid:always
keyUsage               = keyAgreement, keyEncipherment, digitalSignature
#subjectAltName        = @auth_alt_name

# PIV.9c
[ sign ]
basicConstraints       = CA:FALSE
subjectKeyIdentifier   = hash
authorityKeyIdentifier = keyid:always
keyUsage               = digitalSignature, keyEncipherment
extendedKeyUsage       = codeSigning
#subjectAltName        = @sign_alt_name
~~~

**Comment by Jakuje** [comment]
Regarding to the prompts, I think we do not have complete control of the prompts as if it prompts in different contexts, the wording might be a bit different, but it is something we should double-check. 

Regarding to the touch requirement, this is really specific to yubikeys. This is something that is not visible right now on the PKCS#11 level for the pkcs11-provider so if we would like the pkcs11-provider to show this information, we would need to adjusts OpenSC (or your pkcs11 module of choice) to provide custom attribute for a key with this property. We have a PR in progress to support this better in OpenSC/OpenSC#3071, but it is still in progress.

**Comment by simo5** [comment]
@Jakuje I see that https://github.com/OpenSC/OpenSC/pull/3758 has been merged, is there anything exposed at the PKCS#11 elvel that can be used to then handle better prompting?

**Comment by Jakuje** [comment]
> [@Jakuje](https://github.com/Jakuje) I see that [OpenSC/OpenSC#3758](https://github.com/OpenSC/OpenSC/pull/3758) has been merged, is there anything exposed at the PKCS#11 elvel that can be used to then handle better prompting?

No. There is no signaling in pkcs11 that could handle this. We have `sc_notify()` which should show UI dialog depending on the platform/build configuration. In Fedora/RHEL, I do not build notify support because when the opensc is invoked from non-GUI, it did not work well from what I remember.
